src/matchers/toBeValidDate/predicate.js

```diff
@@ -0,0 +1,7 @@
+let is = type => value => Object.prototype.toString.call(value) === `[object ${type}]`;
+
+let hasDateType = is('Date');
+
+let isValidDate = value => hasDateType(value) && !isNaN(value) && !isNaN(value.getTime());
+
+export default expected => isValidDate(expected);
```

Comment:
This can just be:

```js
export default isValidDate;
```

We don't need both the left and right hand sides of the invocation :)